GillianCore/engine/GeneralSemantics/General/GInterpreter.ml

```diff
@@ -587,7 +581,11 @@ struct
                | Some state' -> [ state' ]
                | _ -> [])
              fos)
-    | SpecVar xs -> [ State.add_spec_vars state (Var.Set.of_list xs) ]
+    | FreshSVar x ->
+        let new_svar = Generators.fresh_svar () in
+        let state' = State.add_spec_vars state (SS.singleton x) in
+        let v = State.eval_expr state' (LVar new_svar) in
```

Comment:
Ah, so here the expression is evauated properly (see below)...

Reply:
So, I agree, but Jose's code was saying that he needs to evaluate it and I don't know why 🙃

Reply:
I mean, yes, we can, but in this case the two are equivalent - we should choose one, probably expression evaluation rather than `Val.from_expr`.

Reply:
I decided, it's going to be Val.from_expr. It's cleaner in theory: the lvar is fresh, it always evaluates to itself, and any other interpretation would be wrong.